doc: improve tutorial section

SimeonEhrig opened the pull request:
- be a little bit more verbose in the alpaka::Vec ref documentation
- make the background of the api_deviceKind image transparent

psychocoderHPC commented on docs/snippets/example/00_vector.cpp:
```diff
@@ -159,7 +159,11 @@ TEST_CASE("vector ref", "[docs]")
 
     // creates a permuted view to an existing vector to modify only a subset of components
     concepts::Vector auto vecView = vec0.ref(CVec<uint32_t, 0u, 2u>{});
+    static_assert(vecView.dim() == 2);
```
could you write `2u` else we compare unsigend with signed.

SimeonEhrig replied:
done

SimeonEhrig committed:
doc: improve tutorial section

- be a little bit more verbose in the alpaka::Vec ref documentation
- make the background of the api_deviceKind image transparent
